core/api/services/health_checker.py

```diff
@@ -0,0 +1,19 @@
+from azure.core import exceptions
+from azure.cosmos import CosmosClient
+from api.core.config import STATE_STORE_ENDPOINT, STATE_STORE_KEY
+from api.models.schemas.health import StatusEnum
+from api.resources import strings
+
+
+def get_state_store_status() -> (StatusEnum, str):
```

Comment:
This function has a misleading name. It says get something but instead it creates something. I know we are going to go with connection pool in the next PR but we should perhaps refactor the creation of client into another function which can be get_or_create_connection and then we can replace the functionality of that function.